### ISSUE TITLE
[Dropdown] When `type` of `"hover"` is specified the component cannot be opened via keyboard

### ISSUE THREAD

**Issue by geospatialem** [Esri/calcite-components#13997]
**monday.com sync:** #11395125813

### Check existing issues

- [x] I have [checked for existing issues](https://github.com/Esri/calcite-design-system/issues) to avoid duplicates

### Which Component

Dropdown, Dropdown Group, Dropdown Item

### Actual Behavior

When `dropdown` has a `type` of `"hover"` the component toggles its `open` state when clicked or hovered over. However, the component does not toggle `open` when focused or accessed via the keyboard when in focus.

This differs from `tooltip`, which opens on both hover and focus.

cc @driskull @Esri/calcite-designers 

### Expected Behavior

**Option 1: mitigate/fix**:
- [x] The component is toggled open on click
- [x] The component is toggled open via focus
- [x] The component is toggled open when in focus and selected (via the `Enter` or `spacebar` keys)
- [x] The component's design should remain as-is with different styles for its `:hover` and `:focus` states

**Option 2: deprecate for future removal**:
- [ ] Deprecate the `type` of `"hover"`, with future planned removal

### Reproduction Sample

https://codepen.io/driskull/pen/yyJjqLL

### Reproduction Steps

1. Open the [sample](https://codepen.io/driskull/pen/yyJjqLL)
2. Hover over the component via mouse
3. Observe the component toggles its `open` state
4. Click the component via mouse
5. Observe the component toggles its `open` state
6. Attempt to keyboard navigate into the component without a mouse and attempt to select the component via the `Enter` or `space` keys
7. Ohserve the component's `open` state is not toggled

### Reproduction Version

`3.3.3`, but also applies to `5.0`

### Relevant Info

Relates to [WCAG 1.4.13: Content on Focus or Hover (Level AA)](https://www.w3.org/WAI/WCAG22/Understanding/content-on-hover-or-focus.html):

> ### Goal
> More users can perceive and dismiss non-persistent content.
> ### What to do
> If hover or focus causes content changes, ensure interaction is predictable.
> ### Why it's important
> Unpredictable temporary content can be hard for some to consume and may disrupt others.

### Priority impact

impact - p2 - want for an upcoming milestone

### Impact

The `type="hover"` is not a common workflow, but this is something that conflicts with WCAG criteria that we should provide - consider this medium/high priority to mitigate, or remove its support entirely.

### Calcite package

- [x] @esri/calcite-components
- [ ] @esri/calcite-components-react
- [ ] @esri/calcite-design-tokens
- [ ] @esri/calcite-ui-icons
- [ ] @esri/eslint-plugin-calcite-components

### Esri team

Calcite (dev)

**Comment by macandcheese** [comment]
I'd be in favor of removing `hover` if possible and we don't have any use cases that absolutely require it.

**Comment by geospatialem** [comment]
Awesome, option one is confirmed and verified with `5.1.0-next.15` across states and its design.

cc @driskull @macandcheese @Esri/calcite-designers would we also want to deprecate the `type` of `"hover"` or leave as-is with the fix above?

**Comment by matgalla** [comment]
@ashetland do you think the in progress work with Actions will influence whether or not we keep this?

If not then I'm a +1 for deprecating. Would removal be planned for 7.0 since 6.0 is less than a year out?

**Comment by ashetland** [comment]
Nope. I'm fine with deprecating.

**Comment by geospatialem** [comment]
Closing the above as verified, ~and no follow-up regarding the deprecation of `dropdown` with `type="hover"` with not enough team member buy-in~.

~Will create a new issue if others wish to pursue its deprecation, in particular on the development side.~  

**Update** Created #14173 after additional internal discussions.
